### DISCUSSION TITLE
Change the tests in ocean nightly test suite

### DISCUSSION THREAD

**Discussion by xylar** [MPAS-Dev/compass#188] [General]
I think it makes sense at this point to remove some test cases from the ocean nightly test suite.  Here is the complete list of tests:

- [x] ocean/baroclinic_channel/10km/default
- [x] ocean/baroclinic_channel/10km/threads_test
- [x] ocean/baroclinic_channel/10km/decomp_test
- [x] ocean/baroclinic_channel/10km/restart_test
- [x] ocean/global_ocean/QU240/mesh
- [x] ocean/global_ocean/QU240/PHC/init
- [x] ocean/global_ocean/QU240/PHC/performance_test
- [x] ocean/global_ocean/QU240/PHC/restart_test
- [x] ocean/global_ocean/QU240/PHC/decomp_test
- [x] ocean/global_ocean/QU240/PHC/threads_test
- [x] ocean/global_ocean/QU240/PHC/analysis_test
- [x] ocean/global_ocean/QU240/PHC/RK4/performance_test
- [x] ocean/global_ocean/QU240/PHC/RK4/restart_test
- [x] ocean/global_ocean/QU240/PHC/RK4/decomp_test
- [x] ocean/global_ocean/QU240/PHC/RK4/threads_test
- [ ] ~~ocean/global_ocean/QU240/EN4_1900/init~~
- [ ] ~~ocean/global_ocean/QU240/EN4_1900/performance_test~~
- [ ] ~~ocean/global_ocean/QU240/PHC_BGC/init~~
- [ ] ~~ocean/global_ocean/QU240/PHC_BGC/performance_test~~
- [x] ocean/ice_shelf_2d/5km/z-star/restart_test
- [x] ocean/ice_shelf_2d/5km/z-level/restart_test
- [x] ocean/ziso/20km/default
- [x] ocean/ziso/20km/with_frazil

Checked items are the ones I would keep, strike-through are ones I would remove.

**Comment by xylar** [comment]
cc @vanroekel, @mark-petersen, @sbrus89, @maltrud

**Comment by xylar** [comment]
This might also be a good opportunity to talk about adding more tests, since there are aspects of MPAS-Ocean that aren't covered by these tests.

**Reply by sbrus89** [reply]
It may be good to add one of the wetting and drying tests (dam_break or sloped beach). Perhaps also something that would test the tidal forcing. Maybe one of the single-layer barotropic cases being developed by @mark-petersen and @nairita87 would work for this.

**Reply by vanroekel** [reply]
Thanks for starting the discussion @xylar.  One that comes to mind that I would eventually like to have is some form of the advection convergence test.  As more convergence tests come online through the NGD, I'd like to see those included as well.  Maybe not every night I guess, but at least whenever there is a new PR.

**Reply by sbrus89** [reply]
I second Luke's comment on the advection convergence test.

**Reply by xylar** [reply]
@sbrus89
> It may be good to add one of the wetting and drying tests (dam_break or sloped beach). Perhaps also something that would test the tidal forcing. Maybe one of the single-layer barotropic cases being developed by @mark-petersen and @nairita87 would work for this.

All of these would require development for the compass package that isn't currently being done as far as I know.  So that would be the first step for any of those test cases.

The next is that, as far as I know, many of them were not set up as regression tests.  The require runtimes that are far too long to be useful for that purpose.  So test cases with short forward runs within these test groups would be set up.  This isn't hard.

**Reply by xylar** [reply]
I agree about the convergence testing.  Currently, it's really expensive compared with the rest of the testing so that means we probably need task parallelism for it to become feasible.

**Comment by xylar** [comment]
## EN4_1900

It seems like it makes sense to keep `EN4_1900` versions of test cases around if only to show how an alternative initial condition could be provided.  However, @vanroekel has made clear that `EN4_1900` initial conditions tended to exacerbate E3SM biases rather than help them.  So we don't have any plans to use it for simulations.  It seems silly to have it in the nightly test suite.

**Reply by vanroekel** [reply]
I agree with @xylar makes sense to me to remove the EN4_1900 from the nightly

**Comment by xylar** [comment]
## PHC_BGC

@maltrud has made clear to me in discussions with him that the BGC configuration we use in compass isn't likely to be used by the E3SM BGC team either for regression testing or creating ocean initial conditions.  While it seems like we should have a longer-term goal of moving toward a more unified process, it doesn't make sense to be testing functionality that no one plans to use.
